### PR TITLE
Label utils

### PR DESCRIPTION
**Pull request by mangomadhava**
Not sure if this is the best place to put these functions, thoughts?

check_labels.py: counts files, checks for duplicates in each folder, checks to make sure image-label pairs are all paired up properly. 

count_labels.py: opens each txt file in the bucket and keeps track of the number and type of labels - might be useful to add a date uploaded as well?

download_matched_files.py: Finds and downloads files from street2sat-uploaded that are in a folder in the street2sat-model-predictions

### CODE REVIEW

File: gcp/utils/check_labels.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        '.JPEG' : ''
+    }
+
+    # store the file names
+    file_names = {}
+    for pfx in data_prefixes:
+        files = list(bucket.list_blobs(prefix=pfx))
+        file_names[pfx] = [multiple_replace(dict, x.name) for x in files]
+
+    print('Found:\n {} training images\n {} validation images\n {} training labels\n {} validation labels\n'.format(len(file_names[imgs_train_prefix]), len(file_names[imgs_val_prefix]), len(file_names[labels_train_prefix]), len(file_names[labels_val_prefix ])))
```
Comment (ivanzvonkov) [inline gcp/utils/check_labels.py]: You aren't a fan of fstrings??

File: gcp/utils/count_labels.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        else:
+            counts[int(l[0])] = counts[int(l[0])] + 1
+    return counts
+
+def main():
+    client = storage.Client()
+    bucket = client.get_bucket('street2sat-model-labeled-data')
+
+    labels_train_prefix = 'run2/labels/train/'
+    labels_val_prefix = 'run2/labels/val/'
```
Comment (ivanzvonkov) [inline gcp/utils/count_labels.py]: I suggest putting the above few lines in some common util as similar things are used in the other script

**Comment by ivanzvonkov** [comment]
I think it would be more accurate to call this folder `data_verification` or `data_tests` rather than `utils`

**Comment by hannah-rae** [comment]
Thanks for working on this @mangomadhava and @ivanzvonkov! Let me know when `download_matched_files.py` is ready and I can pull the US data for Dhruv to label too.

File: gcp/utils/check_labels.py (hunk 1 above)
Comment (mangomadhava) [reply gcp/utils/check_labels.py]: Oh true those are great!

File: gcp/data_verification/check_labels.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    # TODO: determine from yaml file
+    imgs_train_prefix = 'run2/images/train/'
+    imgs_val_prefix = 'run2/images/val/'
+    labels_train_prefix = 'run2/labels/train/'
+    labels_val_prefix = 'run2/labels/val/'
+
+    data_prefixes = [imgs_train_prefix, labels_train_prefix, imgs_val_prefix, labels_val_prefix]
+
+    # rules to clean file names
+    dict = {
```
Comment (ivanzvonkov) [inline gcp/data_verification/check_labels.py]: dict is also a default function `dict()`, I suggest using a different name

File: gcp/data_verification/check_labels.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            print(img)
+    if val_extra != set():
+        print("These IMAGES from validation did not have a corresponding label: ")
+        for img in val_extra:
+            print(img)
+
+    # check in labels but not in images
+    train_extra = set(file_names[labels_train_prefix]) - set(file_names[imgs_train_prefix])
+    val_extra = set(file_names[labels_val_prefix]) - set(file_names[imgs_val_prefix])
+    if train_extra != set():
```
Comment (ivanzvonkov) [inline gcp/data_verification/check_labels.py]: `len(train_extra) > 0` might be more clear

File: gcp/data_verification/count_labels.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+def count_from_txt(txt):
+    # labels are stored as integers
+    counts = {}
+    for line in txt.splitlines():
+        l = line.split()
+        if int(l[0]) not in counts.keys():
+            counts[int(l[0])] = 1
+        else:
+            counts[int(l[0])] = counts[int(l[0])] + 1
+    return counts
```
Comment (ivanzvonkov) [inline gcp/data_verification/count_labels.py:13]:
```python
from collections import defaultdict
...
counts = defaultdict(lambda: 0)
for line in txt.splitlines():
        line_int = int(line.split()[0])
        counts[line_int] += 1
```
I think this does the same thing

**Review by ivanzvonkov — APPROVED** [review]
Added some minor suggestions, generally LGTM

**Pull request merged**